tests/framework/extensions/auth/auth.go

```diff
@@ -0,0 +1,34 @@
+package auth
+
+import (
+	"encoding/json"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/stretchr/testify/suite"
+
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+)
+
+// Login uses the local authentication provider to authenticate a user and return the subsequent token.
+func Login(s suite.Suite, c *rancher.Client, user *management.User) (*management.Token, error) {
+	if user.Password == "" {
+		s.Fail("Password cannot be empty")
```

Comment:
Another nitpick sorry, why not just return an error here, instead of failing? We get the error, and fail on the actual test, that way we don't have to pass it the test suite.